runtime/src/accounts.rs

```diff
@@ -118,7 +126,10 @@ impl Accounts {
                 .filter(|key| !message.program_ids().contains(&key))
             {
                 let (account, rent) = AccountsDB::load(storage, ancestors, accounts_index, key)
-                    .and_then(|(account, _)| rent_collector.update(account))
+                    .and_then(|(mut account, _)| {
```

Comment:
I think this needs to move to the store() side...

Reply:
consider the case of account creation (i.e. load returns `Account::default()`, which means it owes rent).  the account receives some lamports, but then we store an account with `rent_epoch` == 0, which means the next time it's loaded, it'll owe a ton of rent.

Reply:
that is, unless the system program charges rent at account creation, which means that `sysvar::rent` *and* `sysvar::clock` need to be passed to create_account, which is un-great.

finally, consider that we probably (per a Discord discussion) want to charge rent for accounts with `data.len()` == 0.  the only good place to do that would be on the store() side.